CI fails on Windows

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Information

- **Qiskit Optimization version**: latest
- **Python version**: 3.8
- **Operating system**: Windows 10

### What is the current behavior?

CI fails. https://github.com/Qiskit/qiskit-optimization/actions/runs/1178358017

```
test.algorithms.test_recursive_optimization.TestRecursiveMinEigenOptimizer.test_recursive_warm_qaoa
---------------------------------------------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "D:\a\qiskit-optimization\qiskit-optimization\test\optimization_test_case.py", line 47, in wrapper
    test_item(self, *args)

      File "D:\a\qiskit-optimization\qiskit-optimization\test\algorithms\test_recursive_optimization.py", line 155, in test_recursive_warm_qaoa
    np.testing.assert_array_almost_equal(cplex_result.x, result.x, 4)

      File "c:\miniconda\lib\site-packages\numpy\testing\_private\utils.py", line 1046, in assert_array_almost_equal
    assert_array_compare(compare, x, y, err_msg=err_msg, verbose=verbose,

      File "c:\miniconda\lib\site-packages\numpy\testing\_private\utils.py", line 844, in assert_array_compare
    raise AssertionError(msg)

    AssertionError: 
Arrays are not almost equal to 4 decimals

Mismatched elements: 2 / 2 (100%)
Max absolute difference: 2.
Max relative difference: 1.
 x: array([0., 2.])
 y: array([2., 0.])
```


### Steps to reproduce the problem



### What is the expected behavior?



### Suggested solutions

--- a/.github/actions/install-main-dependencies/action.yml
+++ b/.github/actions/install-main-dependencies/action.yml
@@ -21,6 +21,10 @@ runs:
   steps:
     - name: Install Dependencies from Main
       run: |
+        if [ "${{ inputs.os }}" == "windows-latest" ]; then
+          source "$CONDA/etc/profile.d/conda.sh"
+          conda activate
+        fi
         pip install git+https://github.com/Qiskit/qiskit-terra
         if [ "${{ inputs.os }}" == "ubuntu-latest" ]; then
           export DISABLE_CONAN=1
@@ -29,8 +33,6 @@ runs:
           sudo apt-get -y install libmuparserx-dev
         fi
         if [ "${{ inputs.os }}" == "windows-latest" ]; then
-          source "$CONDA/etc/profile.d/conda.sh"
-          conda activate
           git clone https://github.com/Qiskit/qiskit-aer.git /tmp/qiskit-aer
           pushd /tmp/qiskit-aer
           python setup.py bdist_wheel -- -G 'Visual Studio 16 2019'

